homeassistant/components/irm_kmi/coordinator.py

```diff
@@ -0,0 +1,137 @@
+"""DataUpdateCoordinator for the IRM KMI integration."""
+
+import asyncio
+from datetime import timedelta
+import logging
+from typing import Any
+
+from irm_kmi_api import IrmKmiApiClientHa, IrmKmiApiError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_LATITUDE, ATTR_LONGITUDE, CONF_ZONE
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import issue_registry as ir
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.update_coordinator import (
+    TimestampDataUpdateCoordinator,
+    UpdateFailed,
+)
+from homeassistant.util import dt as dt_util
+from homeassistant.util.dt import utcnow
+
+from .const import DOMAIN, IRM_KMI_NAME, OUT_OF_BENELUX
+from .data import ProcessedCoordinatorData
+from .utils import disable_from_config, get_config_value, preferred_language
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class IrmKmiCoordinator(TimestampDataUpdateCoordinator):
+    """Coordinator to update data from IRM KMI."""
+
+    def __init__(
+        self, hass: HomeAssistant, entry: ConfigEntry, api_client: IrmKmiApiClientHa
+    ) -> None:
+        """Initialize the coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            config_entry=entry,
+            # Name of the data. For logging purposes.
+            name="IRM KMI weather",
+            # Polling interval. Will only be polled if there are subscribers.
+            update_interval=timedelta(minutes=7),
+        )
+        self._api = api_client
+        self._zone = get_config_value(entry, CONF_ZONE)
+        self.shared_device_info = DeviceInfo(
+            entry_type=DeviceEntryType.SERVICE,
+            identifiers={(DOMAIN, entry.entry_id)},
+            manufacturer=IRM_KMI_NAME.get(
+                preferred_language(self.hass, self.config_entry)
+            ),
+            name=entry.title,
+        )
```

Comment:
rather use a base entity for this

Reply:
I looked at [ecobee](https://github.com/home-assistant/core/blob/dev/homeassistant/components/ecobee/entity.py#L17) to see how it was done.  Implemented in commit fbb5e09712a1c99c3ac0aba3f0b9ba3ef661d17f